tests/sample_qc/test_de_novo.py

```diff
@@ -0,0 +1,244 @@
+"""Test suite for de novo mutation functions."""
+
+import hail as hl
+import pytest
+
+from gnomad.sample_qc.relatedness import (
+    calculate_de_novo_post_prob,
+    default_get_de_novo_expr,
+)
+
+
+class TestDeNovoMutation:
+    """Test suite for de novo mutation functions."""
+
+    @pytest.mark.parametrize(
+        "proband_pl, father_pl, mother_pl, diploid, hemi_x, hemi_y, freq_prior, min_pop_prior, expected",
+        [
+            # Valid test cases (should return expected numeric values)
+            (
+                [73, 0, 161],
+                [0, 99, 198],
+                [0, 99, 198],
+                True,
+                False,
+                False,
+                1.80e-02,
+                100 / 3e7,
+                0.999,
+            ),
+            (
+                [152, 0, 283],
+                [0, 99, 198],
+                [0, 63, 126],
+                True,
+                False,
+                False,
+                7.55e-02,
+                100 / 3e7,
+                0.198,
+            ),
+            # Invalid `freq_prior` case (should raise `HailUserError`)
+            (
+                [99, 50, 0],
+                [0, 99, 198],
+                [0, 99, 198],
+                False,
+                True,
+                False,
+                1.2,
+                100 / 3e7,
+                None,
+            ),
+        ],
+    )
+    def test_calculate_de_novo_post_prob(
```

Comment:
looking at this again, I wonder if this function or the other function should test for missingness in the PL expressions. what do you think?

Reply:
yeah, my thinking was that it might be a good idea to make sure that missing PLs returns a missing value for p_d_novo as expected. fine with me if you think this is covered already